### PR TITLE
fix(config): Fix OkHttpClientConfiguration instantiation

### PR DESCRIPTION
**Pull request by jonsie**
This is a very groovy PR.

### CODE REVIEW

File: clouddriver-eureka/src/main/groovy/com/netflix/spinnaker/config/EurekaProviderConfiguration.groovy [hunk 1]
```diff
@@ -46,6 +46,7 @@ import org.springframework.core.env.ConfigurableEnvironment
 import retrofit.converter.Converter
 import retrofit.converter.JacksonConverter
 
+import javax.inject.Provider
```
Comment (plumpy) [inline clouddriver-eureka/src/main/groovy/com/netflix/spinnaker/config/EurekaProviderConfiguration.groovy]: You don't need this, do you?

**Comment by plumpy** [comment]
Looks like [this is the only other one](https://github.com/spinnaker/kork/blob/7dc4e98d14b476189c88eb17852bf9925e96a430/kork-web/src/main/groovy/com/netflix/spinnaker/config/OkHttpClientComponents.groovy#L44), I think?

File: clouddriver-eureka/src/main/groovy/com/netflix/spinnaker/config/EurekaProviderConfiguration.groovy (hunk 1 above)
Comment (jonsie) [reply clouddriver-eureka/src/main/groovy/com/netflix/spinnaker/config/EurekaProviderConfiguration.groovy]: Ah indeed, left here by mistake.

**Comment by jonsie** [comment]
> Looks like [this is the only other one](https://github.com/spinnaker/kork/blob/7dc4e98d14b476189c88eb17852bf9925e96a430/kork-web/src/main/groovy/com/netflix/spinnaker/config/OkHttpClientComponents.groovy#L44), I think?

This one is actually passing in a `Provider<Registry>` so we're good there.

**Comment by jonsie** [comment]
> Oof, I [fixed this in a Java class in `fiat`](https://github.com/spinnaker/fiat/pull/575) since it caused a compilation error and wondered why it didn't lead to more build breakage... but of course Groovy won't complain.
> 
> I wonder how many more of these are floating out there in Groovy files. We probably need to cherry-pick them all to 1.19.

Yes, this will need to be cherry picked in.  We'll also want kork 7.25.1 in 1.19 as it has a critical fix for OkHttp client configuration properties -- https://github.com/spinnaker/kork/commit/6e885cdd3638358d253fda03cc17ba41504cb333#diff-74a9793617cfd7572ba83c8e96b904d1

**Comment by plumpy** [comment]
@spinnakerbot cherry-pick 1.19

**Pull request merged**